enterprise/dev/ci/internal/buildkite/buildkite.go

```diff
@@ -349,3 +351,14 @@ func AllowDependencyFailure() StepOpt {
 		step.AllowDependencyFailure = true
 	}
 }
+
+// FlattenStepOpts conveniently turns a list of StepOpt into a single StepOpt.
+// It is useful to build helpers that can then be used when defining operations,
+// when the helper wraps multiple stepOpts at once.
+func FlattenStepOpts(stepOpts ...StepOpt) StepOpt {
```

Comment:
I think we can leave this unexported for clarity, seems only used for the cache step

```suggestion
func flattenStepOpts(stepOpts ...StepOpt) StepOpt {
```